### PR TITLE
Enable Linter job in github workflow

### PR DESCRIPTION
**Pull request by sophia-guo**
Fixes: #11505

Note `.github/runlint.sh` is derived from https://github.com/eclipse/openj9/blob/master/buildenv/travis/build-on-travis.sh

### CODE REVIEW

**Comment by sophia-guo** [comment]
Related with #11505

**Comment by sophia-guo** [comment]
Add presubmit.yml and job Linter.  https://github.com/eclipse/openj9/runs/1704368719?check_suite_focus=true

Remove llvm-config and clang++ version check. The version 3.8 is installed specifically and no link is setup automatically between llvm-config to /usr/lib/llvm-3.8/bin/llvm-3.8/llvm-config. Also in travis-ci the default llvm-config and clang++ is 7, which looks like has nothing to do with this lint job?

Only installed g++-7 , gcc-7, nasm, clang-3.8, llvm-3.8, libclang-3.8-dev, llvm-3.8-dev, which looks like the minimum requirements. 

Trigger is set as push and pull-request. Or pull-request is enough?

Naming as presubmit.yml so we could also add build-openj9 and run-aqa functional tests jobs later.

**Comment by smlambert** [comment]
thanks @sophia-guo for kindly creating this draft PR to address #11505 and as a placeholder for us to put some presubmit test targets (@llxia @sophia-guo and Winter 2021 CanOSP team will start to look at change-based testing, first at Adopt, but same approach can eventually apply here).

FYI @eclipse/technology-openj9-committers & interested contributors, please provide feedback, specifically the question regarding triggers.

**Comment by fjeremic** [comment]
> Trigger is set as push and pull-request. Or pull-request is enough?

I think pull-request is enough. It will save some compute cycles against our GHA machine time allocation (if such a thing exists).

**Comment by smlambert** [comment]
@sophia-guo - please remove the push trigger and shift this out of draft state, we'll get it merged and it can be amended by others as they see fit.  👍

File: .github/runlint.sh
Comment (smlambert) [inline .github/runlint.sh:4]: If this is a new file, you can update 2018 to 2021.

**Review by smlambert — APPROVED** [review]
Minor note on copyright date, otherwise LGTM

File: .github/runlint.sh
Comment (sophia-guo) [reply .github/runlint.sh]: Updated. Thanks.

**Comment by pshipton** [comment]
Not a blocker for reviewing/merging this PR, but I noticed the following when running.
```
../compiler/build/scripts/guess-platform.sh: 38: ../compiler/build/scripts/guess-platform.sh: [[: not found
../compiler/build/platform/common.mk:32: PLATFORM not set. Guessing 'amd64-linux64-gcc'
```

**Comment by pshipton** [comment]
Just re-ran the linter job, which you can do from the `Checks`  tab, and it's still passing.

**Comment by keithc-ca** [comment]
> Not a blocker for reviewing/merging this PR, but I noticed the following when running.
> 
> ```
> ../compiler/build/scripts/guess-platform.sh: 38: ../compiler/build/scripts/guess-platform.sh: [[: not found
> ../compiler/build/platform/common.mk:32: PLATFORM not set. Guessing 'amd64-linux64-gcc'
> ```

`[[` is a `bash` feature, which won't be available in lesser shells. `guess-platform.sh` should be fixed - it doesn't need to use `[[`.

File: .github/runlint.sh
Comment (keithc-ca) [inline .github/runlint.sh:44]: Please use `{}` only where necessary (for consistency with the rest of this file).

File: .github/runlint.sh
Comment (keithc-ca) [inline .github/runlint.sh:30]: There is a comment in `build-on-travis.sh` that explains why `4` is chosen.
Do we know the rough specifications of machines that will run this script?

File: .github/runlint.sh
Comment (keithc-ca) [inline .github/runlint.sh:50]: I don't think _all_ of these need to exported.

File: .github/workflows/presubmit.yml
Comment (keithc-ca) [inline .github/workflows/presubmit.yml:53]: Please remove the trailing whitespace

File: .github/runlint.sh
Comment (keithc-ca) [inline .github/runlint.sh:46]: `build-on-travis.sh` included these two commands that I think are still worthwhile:
```
llvm-config --version
clang++ --version
```

**Review by keithc-ca — CHANGES_REQUESTED** [review]

File: .github/runlint.sh
Comment (smlambert) [reply .github/runlint.sh:30]: From https://docs.github.com/en/actions/reference/specifications-for-github-hosted-runners -> https://github.com/actions/virtual-environments/blob/main/images/linux/Ubuntu1604-README.md

Comment (keithc-ca) [reply .github/runlint.sh:30]: According to the second link, jdk11 is available: can we use that rather than downloading another copy?

File: .github/runlint.sh
Comment (sophia-guo) [reply .github/runlint.sh:50]: Could you please suggest which is necessary? Thanks!

File: .github/runlint.sh
Comment (sophia-guo) [reply .github/runlint.sh:46]: Not sure what's the purpose of those two commands? I can see under the travis environment the output is 7.0 and we actually are using 3.8? 

```
+llvm-config --version
7.0.0
+clang++ --version
clang version 7.0.0 (tags/RELEASE_700/final)
```
Looks like clang and llvm 7 are just default travis-ci Compilers and Build toolchain so I guess the two commands were just to verify travis-ci environment?

**Default Compilers and Build toolchain #
clang and llvm 7
cmake 3.12.4
gcc 5.4.0
ccache 3.2.4
shellcheck 0.7.0
shfmt 2.6.3**

https://docs.travis-ci.com/user/reference/xenial/

git-hub runner environment is different from travis-ci. No default llvm and clang is Clang 6.0.0, 8.0.0, 9.0.1.
https://github.com/actions/virtual-environments/blob/main/images/linux/Ubuntu1604-README.md

That means means if we install dependencies by **sudo apt-get -y install clang-3.8 llvm-3.8** and run **llvm-config --version** we will get **llvm-config: command not found**. Unless extra link command to link llvm-config to 3.8

File: .github/runlint.sh
Comment (keithc-ca) [reply .github/runlint.sh:50]: It's disappointing (to me), but perhaps they're all (somewhat) necessary, given the way makefiles were written.

File: .github/runlint.sh
Comment (keithc-ca) [reply .github/runlint.sh:46]: I support the commands would need to be:
```
$LLVM_CONFIG --version
$CXX --version
```
But then, it would only server to confirm what we're getting.

File: .github/runlint.sh
Comment (sophia-guo) [reply .github/runlint.sh:30]: Yes, just check. Same as travis-ci. 2 cores. I will add a similar comment for the chosen reason. 

```
Architecture:          x86_64
CPU op-mode(s):        32-bit, 64-bit
Byte Order:            Little Endian
CPU(s):                2
On-line CPU(s) list:   0,1
Thread(s) per core:    1
Core(s) per socket:    2
Socket(s):             1
NUMA node(s):          1
Vendor ID:             GenuineIntel
CPU family:            6
Model:                 85
Model name:            Intel(R) Xeon(R) Platinum 8171M CPU @ 2.60GHz
Stepping:              4
CPU MHz:               2095.197
BogoMIPS:              4190.39
Hypervisor vendor:     Microsoft
Virtualization type:   full
L1d cache:             32K
L1i cache:             32K
L2 cache:              1024K
L3 cache:              36608K
NUMA node0 CPU(s):     0,1
```

File: .github/runlint.sh
Comment (sophia-guo) [reply .github/runlint.sh]: Updated

File: .github/runlint.sh
Comment (sophia-guo) [reply .github/runlint.sh:31]: Updated the comment and use exising jdk11 - JAVA_HOME_11_X64

File: .github/runlint.sh
Comment (keithc-ca) [inline .github/runlint.sh:42]: The comment is now stale: it can simply be removed.
`SRCDIR` is no longer necessary, because the effect of the two `cd` commands is to go nowhere.

**Review by keithc-ca — APPROVED** [review]

**Comment by keithc-ca** [comment]
@pshipton I'll let you merge this assuming you approve.

**Comment by pshipton** [comment]
Re-ran the linter job and it's still passing.

**Review by pshipton — APPROVED** [review]

**Pull request merged**
